ml-agents/mlagents/trainers/learn.py

```diff
@@ -43,6 +43,31 @@
 logger = logging_util.get_logger(__name__)
 
 
+class DetectDefault(argparse.Action):
```

Comment:
I'm not super happy about this method, but couldn't find another way to detect non-default parameters without removing the `default=` from the `add_argument` calls.

Reply:
Another option would be to remove the defaults / default to `None`.

Reply:
I think the main reason I wanted to keep the defaults was so they would show up when we do `--help`. It'll be especially important once @mmattar removes the command line options documentation. The alternative would be to put the (default=X) text in the help description itself, which isn't terribad either.

Reply:
Makes sense.  I didn't think of the self-documenting side of it.  Adding them in the description text would be OK as long as the description text is generated from the actual default values :-)